lib/compress.js

```diff
@@ -261,6 +262,24 @@ merge(Compressor.prototype, {
                         d.fixed = false;
                     }
                 }
+                if (node instanceof AST_Defun) {
+                    var d = node.name.definition();
+                    if (!toplevel && d.global || is_safe(d)) {
+                        d.fixed = false;
+                    } else {
+                        // Make a shallow copy of definition body but as
+                        // expression instead - ensure to clone whatever
+                        // child nodes you extract from this
```

Comment:
> "ensure to clone whatever child nodes you extract from this"

How can any of the dozens of compress optimizations possibly know it's working on a clone of a tree and can't damage it?